src/puppetlabs/puppetdb/http/factsets.clj

```diff
@@ -1,16 +1,31 @@
 (ns puppetlabs.puppetdb.http.factsets
   (:require [net.cgrand.moustache :refer [app]]
+            [puppetlabs.puppetdb.http :as http]
             [puppetlabs.puppetdb.http.facts :as facts]
             [puppetlabs.puppetdb.http.query :as http-q]
             [puppetlabs.puppetdb.middleware :refer [verify-accepts-json validate-query-params
                                                     wrap-with-paging-options wrap-with-parent-check]]
             [puppetlabs.puppetdb.query.paging :as paging]
-            [puppetlabs.puppetdb.query-eng :refer [produce-streaming-body]]))
+            [puppetlabs.puppetdb.query-eng :as eng]))
+
+(defn factset-status
```

Comment:
Well, stream-query-results is the query engine really, and this is not about appending a restriction to an existing query per se (query= in a singleton case is ignored) it's an absolute query instead. I can see the overlap you perceive though, but I'm not sure it's exactly the same.